stellarphot/core.py

```diff
@@ -62,3 +64,307 @@ def __init__(self, gain=1.0 * u.electron / u.adu,
+
+        # Perform processing once initialized
+        if self.data is None:
+            # Create empty astropy table with right format and data
+            self.data = Table(names=colnames, masked=False, dtype=coltypes, units=colunits)
+        else:
+            # Check the format of the data table matches the table_descript by checking
+            # each column listed in table_descript exists and is he write type and unit.
+            # NOTE: This ignores any columns not in the table_descript, it does not remove them.
+            for i,this_col in enumerate(colnames):
+                # Failure is assumed to be due to bad column value
+                try:
+                    data[this_col]
+                except KeyError:
+                    raise ValueError(f"column {this_col} is not present in input data.")
+                # Check type
+                if data[this_col].dtype != coltypes[i]:
+                    raise ValueError(f"data[{this_col}] is of wrong type (declared {coltypes[i]} but reported as {data[this_col].dtype}).")
+                if data[this_col].unit != colunits[i]:
+                    raise ValueError(f"data[{this_col}] is of wrong unit (declared {colunits[i]} but reported as {data[this_col].unit}).")
+
+        # Create attibutes corresponding to table column
+        self.cols_to_attr(colnames, attrnames)
+
+
+    def cols_to_attr(self, colnames, attrnames):
+        # Connects table columns to attributes
+        for i,attr in enumerate(attrnames):
+            if attr is not None:
+                self.__setattr__(attr, self.data[colnames[i]])
+
+
+class PhotometryData(BaseEnhancedTable):
+    """
+    A base class to hold an `astropy.table.Table` table of reduce photometry
+    data, extending it with a mapping of various table columns to attributes.
+
+    USAGE NOTES: If you input a data file, it MUST contain the following columns
+    in the following column names, types, and units:
+
+    name                 dtype      unit
+    -----------------   -------     -------
+    id                  int64
+    xcenter             float64     pix
+    ycenter             float64     pix
+    aperture_sum        float64     adu
+    annulus_sum         float64
+    RA                  float64     deg
+    Dec                 float64     deg
+    sky_per_pix_avg     float64     adu
+    sky_per_pix_med     float64     adu
+    sky_per_pix_std     float64     adu
+    fwhm_x              float64
+    fwhm_y              float64
+    width               float64
+    aperture float64    pix
+    aperture_area       float64
+    annulus_inner       float64     pix
+    annulus_outer       float64     pix
+    annulus_area        float64
+    exposure            float64       s
+    date-obs            str23
+    night               int64
+    aperture_net_flux   float64     adu
+    BJD                 float64
+    mag_inst            float64
+    airmass             float64
+    filter              str2
```

Comment:
The string types here are problematic because valid values may be longer than the length limit being imposed here. Some of the AAVSO filter names may be more than two characters for example.

I'll look into options here, but my recollection is that there is not a great workaround except making the length large enough to holad any reasonable value.

Reply:
Whatever approach you think is acceptable is fine.

Reply:
Some good news about types: I think we can go with generic python types `float`, `int` and `str`. The `str` column that is created automatically updates the length of the string used to hold the data. The `float` and `int` automatically get converted to numpy dtypes.

This:
```python
t = Table(names=['decimal', 'text', 'integer'], dtype=('float', 'str', 'int'))
```

leads to this:
```
<Table length=0>
decimal text integer
float64 str1  int64
------- ---- -------
```

Reply:
Can you confirm if you subsequently add new data to the `text` column that is longer than the initial allocation, the string length gets updated?